python/dataproc_templates/hive/hive_to_gcs.py

```diff
@@ -139,8 +155,17 @@ def run(self, spark: SparkSession, args: Dict[str, Any]) -> None:
         else:
             output_data = input_data
 
+        #Get partition columns from Hive
+        partition_cols = self.get_hive_partitions(input_table_name,spark)
+        logger.info(
+                "Method get_hive_partitions: received partition columns as: "
+                f"{partition_cols} for hive table {input_table_name}"
+                )
+
         # Write
         writer: DataFrameWriter = output_data.write.mode(output_mode)
+        if not sql_query.strip() and partition_cols:
```

Comment:
Why do we need `not sql_query.strip()` in this condition here?

Reply:
In case a user selects only a few columns when sql_query is used, we might have a situation when partition column is not in selected columns. We could crosscheck columns and then write data onto partitions, but agreed with Hitesh that it overcomplicates the issue. To summarize, we retain partitions by default when moving data from a table directly and not using sql query.